### ISSUE TITLE
wrong-import-order emits multiple identical messages for multiple imports on single line

### ISSUE THREAD

**Issue by RDKRACZ** [pylint-dev/pylint#8808]
### Bug description

The same message is printed 5 times and isn't understandable.
![image](https://github.com/pylint-dev/pylint/assets/43277609/b3b0a40c-1bbf-4232-8404-4c684aa76d14)

### Command used

```shell
pylint ./main.py
```

### Pylint output

```shell
main.py:3:0: C0411: standard import "import requests, re, signal, time, timeit, logging" should be placed before "import requests, re, signal, time, timeit, logging" (wrong-import-order)
main.py:3:0: C0411: standard import "import requests, re, signal, time, timeit, logging" should be placed before "import requests, re, signal, time, timeit, logging" (wrong-import-order)
main.py:3:0: C0411: standard import "import requests, re, signal, time, timeit, logging" should be placed before "import requests, re, signal, time, timeit, logging" (wrong-import-order)
main.py:3:0: C0411: standard import "import requests, re, signal, time, timeit, logging" should be placed before "import requests, re, signal, time, timeit, logging" (wrong-import-order)
main.py:3:0: C0411: standard import "import requests, re, signal, time, timeit, logging" should be placed before "import requests, re, signal, time, timeit, logging" (wrong-import-order)
```

### Expected behavior

Only one `main.py:3:0: C0411: standard import "import requests, re, signal, time, timeit, logging" should be placed before "import requests, re, signal, time, timeit, logging" (wrong-import-order)` that makes sense

### Pylint version

```shell
pylint 2.17.4
astroid 2.15.5
Python 3.10.6 (main, May 29 2023, 11:10:38) [GCC 11.3.0]
```

### OS / Environment

Pop!_OS 22.04 LTS

**Comment by jacobtylerwalls** [comment]
Thanks for the great bug report. The message goes away if you alphabetize "re" before "requests". In this case with a multiple import statement I agree emitting one message is best (five messages that are clearer would be acceptable too, but probably harder to implement and less intuitive.)

**Comment by azinneck0485** [comment]
Can I work on this?  I've figured out a way to possibly implement a clearer message to something like:
`standard import "re" should be placed before "requests" (wrong-import-order)`
if the import is `import request, re`

I need to refine the logic to handle the example case in the issue, where there are more than 2 imports.  It's possible that this can then be leveraged to avoid repeating messages, but that will need further investigation.

**Comment by jacobtylerwalls** [comment]
Of course! Thanks for lending a hand 🤝

**Comment by Pierre-Sassoulas** [comment]
I'm not sure it's easy to do, we use isort to raise this message or not so we'd need to re-implement the logic in pylint with all the false positives that can happen if we disagree with isort. (And we could use ruff in the future instead of isort, complicating things further)

**Comment by jacobtylerwalls** [comment]
Oh I definitely agree we shouldn't reimplement isort. I just figured some light deduplication of the resulting messages was the ticket.

**Comment by azinneck0485** [comment]
After taking a look at the code, I made some changes in the strings provided to the wrong-import-message string to clarify what the message trigger.  Now for the imports in the initial issue, pylint will output:

`test.py:7:0: C0410: Multiple imports on one line (requests, re, signal, time, timeit, logging) (multiple-imports)
test.py:7:0: C0411: standard import "re" should be placed before "requests" (wrong-import-order)
test.py:7:0: C0411: standard import "signal" should be placed before "requests" (wrong-import-order) 
test.py:7:0: C0411: standard import "time" should be placed before "requests" (wrong-import-order)   
test.py:7:0: C0411: standard import "timeit" should be placed before "requests" (wrong-import-order) 
test.py:7:0: C0411: standard import "logging" should be placed before "requests" (wrong-import-order)
`

Instead of 5 identical messages, each one specifies the offending import and where it should be placed.  If an import is out of order relative to several, the message ends with `...should be placed before "import1, import2"` with the hope of streamlining changes needed to clear the message.

Is this along the lines of what would be expected to address this issue or should these also be consolidated in a single message (i.e., add logic to suggest the correct order of imports *when on a single line*) to add brevity?

**Comment by Pierre-Sassoulas** [comment]
Yes I think it's better ! There's a problem in the logic though because ``time`` should be placed after ``requests`` not before ?

**Comment by azinneck0485** [comment]
> Yes I think it's better ! There's a problem in the logic though because `time` should be placed after `requests` not before ?

I'm not sure.  From what I can tell, the code is checking the category returned from `isort`'s add_module and using that to check import order, so it is seeing `time` as a standard import and `requests` as a third-party import, so `time` should be before `requests` based on that.

I think before I made the change, I couldn't reproduce the finding [here](https://github.com/pylint-dev/pylint/issues/8808#issuecomment-1615895763) (I need to double check and will do that).  So is it possible that there is something wrong or missing in the logic if this is the correct or expected behavior.

I will check more into the code and previous behavior compared to new behavior.  Is there a certain rule or convention that should be part of the logic to enforce this ordering, if it is not already present?

**Comment by Pierre-Sassoulas** [comment]
> I'm not sure. From what I can tell, the code is checking the category returned from isort's add_module and using that to check import order, so it is seeing time as a standard import and requests as a third-party import, so time should be before requests based on that.

Ha, you're right, request is a third party library not time.  Maybe adding the reason in parenthesis would be nice but keeping it simple and stupid is nice too. (Disclaimer: I'm not using this message myself because isort/ruff can automate the fix).

**Comment by azinneck0485** [comment]
> > I'm not sure. From what I can tell, the code is checking the category returned from isort's add_module and using that to check import order, so it is seeing time as a standard import and requests as a third-party import, so time should be before requests based on that.
> 
> Ha, you're right, request is a third party library not time. Maybe adding the reason in parenthesis would be nice but keeping it simple and stupid is nice too. (Disclaimer: I'm not using this message myself because isort/ruff can automate the fix).

No worries.  Just wanted to make sure I wasn't missing something :-)  I will improve the message to provide a bit more information.

I see that PEP8 provides the "grouping" guidance for imports, but does not specify they need to be alphabetical.  I know isort does it this way, so I was wondering if that is something that pylint should check (in a new issue, if one doesn't exist yet)?

**Comment by azinneck0485** [comment]
Made changes to the message to separate out each category of imports:
`standard import "re" should be placed before third party import(s) "requests" (wrong-import-order)`

In the case imports of multiple categories are out of order, the last part reads something like `...third party import(s) "requests" and first party import(s) "pylint"`.  This is more helpful to me than my original proposal, but others may find it too much information.

The only other change I can think that could be helpful is to reference PEP8 (as a parenthetical) in the message, but that may be redundant since the documentation for wrong-import-order already states this.

Any thoughts on what would be most helpful for a general user (maybe something I haven't considered yet)?

**Comment by Pierre-Sassoulas** [comment]
I think ``standard import "re" should be placed before third party import(s) "requests" (wrong-import-order)`` is optimal, no need to tell about pep8 in the message displayed, but there's documentation surrounding the message that you can upgrade if required (https://pylint.readthedocs.io/en/stable/user_guide/messages/convention/wrong-import-order.html)
